Remove 'withdraw' functionality and convert SendEther to work with BPTXM (#3060)
smartcontractkit/chainlink: CHANGELOG-core.md

- New prometheus metrics for the head tracker:
  - `head_tracker_heads_in_queue` - The number of heads currently waiting to be executed. You can think of this as the 'load' on the head tracker. Should rarely or never be more than 0.
  - `head_tracker_callback_execution_duration` - How long it took to execute all callbacks. If the average of this exceeds the time between blocks, your node could lag behind and delay transactions.

> - `ethtx` tasks now support a new parameter, `minRequiredOutgoingConfirmations` which allows you to tune how many confirmations are required before moving on from an `ethtx` task on a per task basis (only works with BulletproofTxManager). If it is not supplied, the default of `MIN_OUTGOING_CONFIRMATIONS` is used (same as the old behaviour).

Reviewer: I don't understand the difference in behaviour between `minRequiredOutgoingConfirmations` and `MIN_OUTGOING_CONFIRMATIONS`?

From what I can read in this diff https://github.com/smartcontractkit/chainlink/pull/2997/files#diff-b97fb224c64cd286dcf343e0bcae3203R423-R425 they do the same thing, one is global and one is configurable per task. 

Can they have the same name then?! so `minOutgoingConfirmations`?

Author: `minRequiredOutgoingConfirmations` is a more descriptive name IMO.

We cannot rename `MIN_OUTGOING_CONFIRMATIONS` because it is a historically used env var. Two options:

1. Rename `minOutgoingConfirmations()` to `minRequiredOutgoingConfirmations()` (but keep the old env var name
2. Rename `minRequiredOutgoingConfirmations` to `minOutgoingConfirmations` everywhere

Which do you prefer?

Reviewer: 1.

- `ETH_FINALITY_DEPTH` specifies how deep protection should be against re-orgs. The default is 50. It only applies if BulletproofTxManager is enabled. It is not recommended to change this setting.
- `EthHeadTrackerHistoryDepth` specifies how many heads the head tracker should keep in the database. The default is 100. It is not recommended to change this setting.

### Breaking changes

> - `admin withdraw` command has been removed. This was only ever useful to withdraw LINK if the Oracle contract was owned by the Chainlink node address. It is no longer recommended to have the Oracle owner be the chainlink node address.

Reviewer 2: ```suggestion
- `admin withdraw` command has been removed. This was only ever useful to withdraw LINK if the Oracle contract was owned by the Chainlink node address. It is recommended to have the Oracle owner be a different address than the Chainlink node address.
```
It was never recommended to have the Oracle owner be the Chainlink node address, it was only available as an option.

Author: Thanks @thodges-gh

> ### Breaking changes
> 
> - `admin withdraw` command has been removed. This was only ever useful to withdraw LINK if the Oracle contract was owned by the Chainlink node address. It is recommended to have the Oracle owner be a different address than the Chainlink node address.
> - Fixed `txs create` to send the amount in Eth not in Wei (as per the documentation)
> 

Reviewer 3: ```suggestion
### Changed

- Fixed `txs create` to send the amount in Eth not in Wei (as per the documentation)

### Removed

- `admin withdraw` command has been removed. This was only ever useful to withdraw LINK if the Oracle contract was owned by the Chainlink node address. It is recommended to have the Oracle owner be a different address than the Chainlink node address.

```

We're trying to follow the [KeepAChangelog](https://keepachangelog.com/en/1.0.0/) format, which supports 6 categories(Added, Changed, Deprecated, Removed, Fixed, and Security). I hope that sticking to these categories will make it easier for readers to parse out what they care about. (I missed the `Env var changes section` last time, but would ideally like to roll that into this categorization as well.

Author: Didn't know that, nice :+1:
